### PR TITLE
benchmark: add sampling.renorm benchmarks

### PR DESCRIPTION
**Pull request by xslingcn**
Partial results:
```
---
top-p renorm
vocab_size: 128512, batch_size: 512, distrib: normal_distribution(std=1), p: 0.1, duration: 2014.94 us, effective bandwidth: 261.24 GB/s
vocab_size: 128512, batch_size: 512, distrib: normal_distribution(std=1), p: 0.9, duration: 2568.39 us, effective bandwidth: 204.95 GB/s
vocab_size: 128512, batch_size: 512, distrib: gumbel_distribution(beta=1), p: 0.1, duration: 2021.48 us, effective bandwidth: 260.40 GB/s
vocab_size: 128512, batch_size: 512, distrib: gumbel_distribution(beta=1), p: 0.9, duration: 2377.93 us, effective bandwidth: 221.36 GB/s
---
top-k renorm
vocab_size: 128512, batch_size: 512, distrib: normal_distribution(std=1), k: 10, duration: 1761.15 us, effective bandwidth: 298.89 GB/s
vocab_size: 128512, batch_size: 512, distrib: normal_distribution(std=1), k: 100, duration: 2299.25 us, effective bandwidth: 228.94 GB/s
vocab_size: 128512, batch_size: 512, distrib: gumbel_distribution(beta=1), k: 10, duration: 1830.94 us, effective bandwidth: 287.49 GB/s
vocab_size: 128512, batch_size: 512, distrib: gumbel_distribution(beta=1), k: 100, duration: 2286.19 us, effective bandwidth: 230.25 GB/s
---
top-k mask logits
vocab_size: 128512, batch_size: 512, distrib: normal_distribution(std=1), k: 10, duration: 1401.01 us, effective bandwidth: 375.72 GB/s
vocab_size: 128512, batch_size: 512, distrib: normal_distribution(std=1), k: 100, duration: 1661.22 us, effective bandwidth: 316.87 GB/s
vocab_size: 128512, batch_size: 512, distrib: gumbel_distribution(beta=1), k: 10, duration: 1551.09 us, effective bandwidth: 339.36 GB/s
vocab_size: 128512, batch_size: 512, distrib: gumbel_distribution(beta=1), k: 100, duration: 1798.39 us, effective bandwidth: 292.70 GB/s
```

### CODE REVIEW

File: benchmarks/bench_renorm.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        U = torch.rand(shape, device=device)
+        eps = 1e-20
+        return torch.log(-torch.log(U + eps) + eps) / beta
+
+    gumbel_noise.__name__ = f"gumbel_distribution(beta={beta})"
+    return gumbel_noise
+
+
+def init_seed_top_p_renorm(*args, **kwargs):
+    torch.manual_seed(42)
```
Comment (yzh119) [inline benchmarks/bench_renorm.py]: We don't really to fix seed here because there is no randomness in renorm kernels. Please move them to the front of `main()` to fix `distrib`.

File: benchmarks/bench_renorm.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    return flashinfer.sampling.top_p_renorm_probs(*args, **kwargs)
+
+
+def init_seed_top_k_renorm(*args, **kwargs):
+    torch.manual_seed(42)
```
Comment (yzh119) [inline benchmarks/bench_renorm.py]: ditto

File: benchmarks/bench_renorm.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    return flashinfer.sampling.top_k_renorm_probs(*args, **kwargs)
+
+
+def init_seed_top_k_mask_logits(*args, **kwargs):
+    torch.manual_seed(42)
```
Comment (yzh119) [inline benchmarks/bench_renorm.py]: ditto.
